File: Searching and Sorting/Sorting Algo/SelectionSort.cpp
#include <iostream>
using namespace std;

void SelectionSort(int arr[], int size){
    int temp;
    for(int i=0;i<size-1;i++){
        for(int j=i+1;j<size;j++){
            if(arr[i]>arr[j]){   //Here you can create minIndex var and store that, just updating that min index find min value from remaining array. By this approach you need to only swap element once.
                temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
            }
        }
    }
    cout<<"Sorted Array: ";
    for(int i=0;i<size;i++){
        cout<<arr[i]<<" ";
    }

    return;

}

int main(){
    int arr[] = {20,15,30,39,9,3,12,34};
    int size = sizeof(arr)/sizeof(arr[0]);
    SelectionSort(arr,size);

    return 0;
}

File: Searching and Sorting/Sorting Algo/selectionSort.py

def selectionSort(arr):
    for i in range(len(arr)-1):
        min = float('inf')
        index = 0
        # print(f"i={i}")
        for j in range(i+1, (len(arr))):
            # print(f"j={j}")
            if arr[j]<min:
                min = arr[j]
                index = j
        
        print(f"i={i} min={min}")
        if min<arr[i]:
            temp = arr[i]
            arr[i] = min
            arr[index] = temp
            
        
    return arr
            
            
        
        



##### main ###
if __name__ == "__main__":
    print("Selection Sort")
    
    arr = [7,5,9,2,8]
    
    sorted_array = selectionSort(arr)
    
    print(f"Sorted Array \n {sorted_array}")